cpp/src/utilities/cudf_utils.h

```diff
@@ -19,15 +20,15 @@
 #endif
 
 inline gdf_error set_null_count(gdf_column* col) {
-  gdf_size_type valid_count{};
-  gdf_error result =
-      gdf_count_nonzero_mask(col->valid, col->size, &valid_count);
-
-  GDF_REQUIRE(GDF_SUCCESS == result, result);
-
-  col->null_count = col->size - valid_count;
-
-  return GDF_SUCCESS;
+    assert(col != nullptr);
```

Comment:
In line with https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#i6-prefer-expects-for-expressing-preconditions I think we're going to move away from using `assert` and instead state preconditions with `GDF_REQUIRES` (deprecated) and `CUDF_EXPECTS`.

Reply:
So shall I make this a void function then, using only `CUDF_EXPECTS()`?

Reply:
So shall I make this a void function then, using only `CUDF_EXPECTS()`?

Reply:
Another option, which I like better, is for set_null_count to take a `gdf_column&`, so that nullness is a non-issue in this funtion - and it's the callers' problem if they create pointers to null `gdf_column`'s (which they shouldn't obviously).

Reply:
> Another option, which I like better, is for set_null_count to take a `gdf_column&`, so that nullness is a non-issue in this funtion - and it's the callers' problem if they create pointers to null `gdf_column`'s (which they shouldn't obviously).

That works for me.

Reply:
Done. The new `CUDF_TRY()` macro we were discussing went into the fix for #1369  in PR #1386.